Sources/StreamChat/Database/DTOs/AttachmentDTO.swift

```diff
@@ -15,17 +15,19 @@ class AttachmentDTO: NSManagedObject {
     }
 
     /// An attachment type.
-    @NSManaged private var type: String
+    @NSManaged private var type: String?
     var attachmentType: AttachmentType {
-        get { .init(rawValue: type) }
+        get { AttachmentType(rawValue: type ?? AttachmentType.unknown.rawValue) }
         set { type = newValue.rawValue }
     }
 
     /// An attachment local state.
-    @NSManaged private var localStateRaw: String
+    @NSManaged private var localStateRaw: String?
     @NSManaged private var localProgress: Double
     var localState: LocalAttachmentState? {
```

Comment:
Not at the moment, in fact the setter applies an empty string, which has no mapping in the LocalAttachmentState enum. So this could be the unknown marker.

Reply:
Should we add that .unknown case then?

Reply:
Pushed a change on your feedback.